src/ckeditor.js

```diff
@@ -0,0 +1,133 @@
+/**
+ * @license Copyright (c) 2003-2019, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* global CKEDITOR */
+
+import getEditorNamespace from './utils/get-editor-namespace.js';
+
+export default {
+	name: 'ckeditor',
+
+	render( createElement ) {
+		return createElement( 'div', {}, [
+			createElement( this.tagName )
+		] );
+	},
+
+	props: {
+		value: {
+			type: String,
+			default: ''
+		},
+		type: {
+			type: String,
+			default: 'classic',
+			validator: type => type in { classic: 1, inline: 1 }
+		},
+		editorUrl: {
+			type: String,
+			default: 'https://cdn.ckeditor.com/4.11.4/standard/ckeditor.js'
+		},
+		config: {
+			type: Object,
+			default: () => {}
+		},
+		tagName: {
+			type: String,
+			default: 'textarea'
+		},
+		readOnly: {
+			type: Boolean,
+			default: null // Use the null for default value, so `config.readOnly` can take precedence.
+		}
+	},
+
+	readOnly() {
+		return this.readOnly === null
+			? this.instance.readOnly
+			: this.readOnly;
+	},
+
+	mounted() {
+		getEditorNamespace( this.editorUrl ).then( () => {
+			const editor = this.instance =
+				CKEDITOR[ this.type === 'inline' ? 'inline' : 'replace' ]( this.$el.firstElementChild, this.config );
+
+			editor.on( 'instanceReady', () => {
+				if ( this.readOnly !== null ) {
+					this.instance.setReadOnly( this.readOnly );
+				}
+
+				const undo = editor.undoManager;
+				const data = this.value;
+
+				undo && undo.lock();
+
+				editor.setData( data, { callback: () => {
+					this.$_setUpEditorEvents();
+
+					// Locking undoManager prevents 'change' event.
+					// Trigger it manually to update bound data.
+					if ( data !== editor.getData() ) {
+						editor.once( 'change', () => {
+							this.$emit( 'ready', editor );
+						} );
+
+						editor.fire( 'change' );
```

Comment:
>editor#change is used to emit component#input, as data can be updated by ACF, we want to emit proper event. Firing component#input manually wouldn't be consisted with how normally it is fired with an original change evt passed as a param.

It's editor initialization code, so when you are using CKEditor (without integration) you won't get `change` event when filling up an editor with data from DOM element (the one you initialize editor on). If we treat initialization `value` with the same purpose, it also shouldn't emit `change` event when mounting component.

Moreover, you are locking `undo` feature before firing `change` event and `change` event is introduced by `undo` plugin to handle creating undo snapshots, which doesn't make sense for me. It looks like workaround over the issue with synchronizing v-model between editor and vue, although why not use mentioned `$emit#input` directly?

Reply:
>Moreover, you are locking undo feature before firing change event and change event is introduced by undo plugin to handle creating undo snapshots, which doesn't make sense for me. It looks like workaround over the issue with synchronizing v-model between editor and vue, although why not use mentioned $emit#input directly?

We can use `$emit`, but undo needs to be locked so there is no initial undo step from call on `setData`.